test/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/initialgainorloss/InitialGainOrLossControllerSpec.scala

```diff
@@ -0,0 +1,272 @@
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.Generators.{sample, _}
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.JourneyStatus.FillingOutReturn
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.finance.AmountInPence
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.returns.InitialGainOrLossAnswers.CompleteInitialGainOrLossAnswers
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.returns.{DisposalDate, DraftReturn}
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.{Error, SessionData}
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.repos.SessionStore
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.services.returns.ReturnsService
+
+import scala.concurrent.Future
+
+class InitialGainOrLossControllerSpec
+    extends ControllerSpec
+    with AuthSupport
+    with SessionSupport
+    with ReturnsServiceSupport
+    with ScalaCheckDrivenPropertyChecks
+    with RedirectToStartBehaviour {
+
+  implicit lazy val messagesApi: MessagesApi = controller.messagesApi
+
+  override val overrideBindings = List[GuiceableModule](
+    bind[AuthConnector].toInstance(mockAuthConnector),
+    bind[SessionStore].toInstance(mockSessionStore),
+    bind[ReturnsService].toInstance(mockReturnsService)
+  )
+
+  lazy val controller                  = instanceOf[InitialGainOrLossController]
+  implicit lazy val messages: Messages = MessagesImpl(Lang("en"), messagesApi)
+  val initialGainOrLosses              = sample[AmountInPence].copy(value = 2)
+
+  def sessionWithState(
+    answers: CompleteInitialGainOrLossAnswers,
+    disposalDate: DisposalDate
+  ): (SessionData, FillingOutReturn) = {
+    val journey = sample[FillingOutReturn].copy(
+      draftReturn = sample[DraftReturn].copy(
+        initialGainOrLossAnswers = Some(answers)
+      )
+    )
+
+    SessionData.empty.copy(journeyStatus = Some(journey)) -> journey
+  }
+
+  val allQuestionsAnswered: CompleteInitialGainOrLossAnswers = CompleteInitialGainOrLossAnswers(
+    initialGainOrLosses
+  )
+
+  val (session, journey) = sessionWithState(allQuestionsAnswered, sample[DisposalDate])
+  val updatedDraftReturn =
+    journey.draftReturn.copy(initialGainOrLossAnswers = Some(CompleteInitialGainOrLossAnswers(AmountInPence(0))))
+  val updatedSession = session.copy(journeyStatus = Some(journey.copy(draftReturn = updatedDraftReturn)))
+
+  "InitialGainOrLossController" should {
+    "show display" when {
+      def performAction(): Future[Result] = controller.enterInitialGainOrLoss()(FakeRequest())
+      "display a page returning 200" in {
+
+        val fillingOutReturn = sample[FillingOutReturn]
+
+        inSequence {
+          mockAuthWithNoRetrievals()
+          mockGetSession(
+            SessionData.empty.copy(
+              journeyStatus = Some(fillingOutReturn)
+            )
+          )
+        }
+
+        val result: Future[Result] = performAction()
+        status(result) shouldBe OK
```

Comment:
you can use `checkPageIsDisplayed` here and in the rest of this spec to help do some more assertions on the page. It allows you to select parts of the html using JSoup to perform assertions on. You can assert things like submit button href's and backlinks. You can also assert that certain radio buttons are preselected if you've already answered the question. There are examples in the other return controller specs

Reply:
Yes, Done.